lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -496,8 +495,6 @@ pub async fn handle_transferred_shard_proxy(
     to: PeerId,
     sync: bool,
 ) -> CollectionResult<bool> {
-    // TODO: Ensure cancel safety!
-
```

Comment:
I don't think this is resolved, cause there's `replica_set.remove_local().await` below that may not be cancel safe, and so `handle_transferred_shard_proxy` may not be cancel safe.

Reply:
Fair enough, but the same TODO is still there a few lines below.

Reply:
My eye now catches the `add_remote` line which may be troublesome as well. 

Alright, lets put it back. Sorry.

Reply:
> Fair enough, but the same TODO is still there a few lines below.

My bad. It's how I format these things. I put two of these `TODO`s, one at the top, and one at a particular thing that need work (if there's a specific item identified).